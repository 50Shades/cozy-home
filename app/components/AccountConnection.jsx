/** @jsx h */
import { h, Component } from 'preact'
import statefulForm from '../lib/statefulForm'

import AccountLoginForm from './AccountLoginForm'
import DataItem from './DataItem'
import ReactMarkdown from 'react-markdown'

class AccountConnection extends Component {
  constructor (props, context) {
    super(props, context)

    this.state = {
      submitting: false
    }

    this.submit = () => {
      this.setState({
        submitting: true
      })

      this.props.submit()
    }
  }

  render ({ t, connector, fields, dirty, error, submit }) {
    const { name, customView, description } = connector
    const { submitting } = this.state
    // If there is no field displayed, the form is dirty by default.
    dirty = dirty || Object.values(fields).every(field => field.type === 'hidden' || field.advanced)
    return (
      <div class='account-connection'>
        <div class='account-description'>
          <h3>{t('my_accounts title description')}</h3>
          <p>
            <ReactMarkdown
              source={
                t(description)
              }
              renderers={{Link: props => <a href={props.href} target='_blank'>{props.children}</a>}}
            />
          </p>
          <h3>{t('dataType title')}</h3>
          <ul class='account-datas'>
            {connector.dataType && connector.dataType.map(data =>
              <DataItem
                dataType={data}
                hex={connector.color.hex}
              />
            )}
          </ul>
          <p>{` ${connector.name} ${t('dataType disclaimer')} `}</p>
        </div>
        <div class='account-login'>
          <h3>{t('my_accounts account config title', {name: name})}</h3>
          <div class={'account-form' + (error ? ' error' : '')}>
            <AccountLoginForm
              t={t}
              customView={customView}
              fields={fields}
            />
            <div class='account-form-controls'>
              <button
                disabled={!dirty}
                aria-busy={submitting ? 'true' : 'false'}
                onClick={() => this.submit()}
              >
                {t('my_accounts account config button')}
              </button>
              {error === 'bad credentials' &&
                <p class='errors'>{t('my_accounts account config bad credentials')}</p>
              }
            </div>
          </div>
        </div>
      </div>
    )
  }
}

export default statefulForm()(AccountConnection)
