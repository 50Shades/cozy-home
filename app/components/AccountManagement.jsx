/** @jsx h */
import { h } from 'preact'
import TimeAgo from 'timeago-react'
import statefulForm from '../lib/statefulForm'

import AccountConfigForm from './AccountConfigForm'

const AccountManagement = (props) => {
  const { t, locale, accounts, selectedAccount, lastImport, dirty, submit, cancel } = props
  const { submitting, synching, deleting } = props
  const { selectAccount, addAccount, synchronize, deleteAccount } = props
  const isLoginFilled = !!props.values.login
  return (
    <div>
      <div class='account-management'>
        <div class='account-list'>
          <ul>
            {accounts.map((account, key) => (
              <li class={selectedAccount === key ? 'selected' : ''}>
                <a onClick={() => selectAccount(key)}>
                  {account.login
                    ? account.login
                    : t('data_connect account index', {index: key + 1})}
                </a>
              </li>
            ))}
          </ul>
          <a className='add-button' onClick={() => addAccount()}>{t('data_connect add_account button')}</a>
        </div>
        <div class='account-config'>
          <div>
            <h3>{t('data_connect activity')}</h3>
            <p>
              {t('data_connect activity desc')}
              {synching
                ? t('data_connect activity running')
                : lastImport && <TimeAgo datetime={lastImport} locale={locale} />
              }
            </p>
            <button class='flat' disabled={synching} onClick={() => synchronize()}>
              {t('data_connect activity button')}
            </button>
          </div>
          <AccountConfigForm {...props} />
          {isLoginFilled
          ? <div>
            <h3>{t('data_connect disconnect')}</h3>
            <p>
              {t('data_connect disconnect desc')}
            </p>
            <button class='danger' disabled={deleting} onClick={() => deleteAccount(selectedAccount)}>
              {t('data_connect disconnect button')}
            </button>
          </div>
          : <div>
            <button class='danger' disabled={deleting} onClick={() => deleteAccount(selectedAccount)}>
              {t('data_connect delete button')}
            </button>
          </div>}

        </div>
      </div>
      <div class='account-management-controls'>
        <button class='cancel' onClick={cancel}>
          {t('data_connect account cancel button')}
        </button>
        <button
          disabled={!dirty}
          aria-busy={submitting ? 'true' : 'false'}
          onClick={submit}
        >
          {t('data_connect account save button')}
        </button>
      </div>
    </div>
  )
}

export default statefulForm()(AccountManagement)
