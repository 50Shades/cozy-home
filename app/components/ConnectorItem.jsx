/** @jsx h */
import { h } from 'preact'
import { Link, withRouter } from 'react-router'
import { translate } from '../plugins/preact-polyglot'

const ConnectorItem = ({ title, subtitle, connected, slug, iconName, enableDefaultIcon = false, router }) => (
  <Link class='item-wrapper' to={`${router.location.pathname}/${slug}`}>
    <header class='item-header'>
      {iconName &&
        <svg class='item-icon'>
          <use xlinkHref={icon(iconName, enableDefaultIcon)} />
        </svg>
      }
    </header>
    <p class='item-title'>{title}</p>
    {subtitle && <p class='item-subtitle'>{subtitle}</p>}
    {connected &&
      <svg class='item-connected'>
        <use xlinkHref={require('../assets/sprites/icon-check.svg')} />
      </svg>
    }
  </Link>
)

// Fallback to get the item icon and avoid error if not found
// with a possible default icon
const icon = (iconName, enableDefaultIcon) => {
  let icon = ''
  try {
    icon = require(`../assets/icons/color/${iconName}.svg`)
  } catch (e) {
    if (enableDefaultIcon) {
      icon = require('../assets/icons/color/default_myaccount.svg')
    }
  }
  return icon
}

export default translate()(withRouter(ConnectorItem))
