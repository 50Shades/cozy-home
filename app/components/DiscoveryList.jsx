/** @jsx h */
import { h } from 'preact'
import { translate } from '../plugins/preact-polyglot'
import UseCaseList from './UseCaseList'

const DiscoveryList = ({ t, useCases, context, children }) => (
  <div class='content'>
    <h1>{t('data_connect discovery title')}</h1>
    <UseCaseList useCases={useCases} context={context} />
    {children}
  </div>
)

export default translate()(DiscoveryList)
