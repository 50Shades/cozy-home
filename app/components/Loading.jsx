import React from 'react'
import { translate } from '../plugins/preact-polyglot'

export const Loading = ({ t, loadingType, noMargin }) => {
  return (
    <div
      className={noMargin
        ? 'coz-loading--no-margin'
        : 'coz-loading'
      }
    >
      {loadingType && <p>{t(`Loading.${loadingType}`)}</p>}
    </div>
  )
}

export default translate()(Loading)
