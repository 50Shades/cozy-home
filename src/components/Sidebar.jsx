import React from 'react'
import { Link, withRouter } from 'react-router'
import { translate } from 'cozy-ui/react/I18n'

export const Sidebar = ({ t, categories, router }) => {
  let isCategoryView = router.location.pathname.match(/^\/providers/) !== null
  const i18nCategories = categories
    .filter(c => c !== 'others')
    .map(c => ({ slug: c, label: t(`category.${c}`) }))
    .sort((a, b) => a.label.localeCompare(b.label))

  return (
    <aside className="coz-sidebar">
      <nav>
        <ul className="coz-nav">
          <li className="coz-nav-item">
            <Link
              to="/connected"
              className="coz-nav-link col-cat-connected"
              activeClassName="active"
            >
              {t('nav.connected')}
            </Link>
          </li>
          <li className="coz-nav-item">
            <Link
              to="/providers/all"
              className={
                isCategoryView
                  ? 'coz-nav-link col-cat-categories active'
                  : 'coz-nav-link col-cat-categories'
              }
            >
              {t('nav.providers')}
            </Link>
          </li>
          {isCategoryView && (
            <ul className="col-nav-submenu">
              <li>
                <Link
                  to="/providers/all"
                  className="col-nav-submenu-link"
                  activeClassName="active"
                >
                  {t('category.all')}
                </Link>
              </li>
              {i18nCategories.map(category => (
                <li>
                  <Link
                    to={`/providers/${category.slug}`}
                    className="col-nav-submenu-link"
                    activeClassName="active"
                  >
                    {category.label}
                  </Link>
                </li>
              ))}
              <li>
                <Link
                  to="/providers/others"
                  className="col-nav-submenu-link"
                  activeClassName="active"
                >
                  {t('category.other')}
                </Link>
              </li>
            </ul>
          )}
        </ul>
      </nav>
    </aside>
  )
}
export default translate()(withRouter(Sidebar))
