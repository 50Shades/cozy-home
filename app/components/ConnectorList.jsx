/** @jsx h */
import { h } from 'preact'
import { translate } from '../plugins/preact-polyglot'
import ConnectorItem from './ConnectorItem'

const ConnectorList = ({ t, connectors, showConnectedBadge = true }) => (
  <div className='connector-list'>
    {connectors.map(c =>
      <ConnectorItem
        title={c.name}
        subtitle={t(c.category + ' category')}
        connected={showConnectedBadge && c.accounts.length !== 0}
        iconName={c.slug}
        slug={c.slug}
        enableDefaultIcon
        backgroundCSS=''
      />
    )}
  </div>
)

export default translate()(ConnectorList)
