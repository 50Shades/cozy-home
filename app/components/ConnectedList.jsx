import React from 'react'
import { translate } from '../plugins/preact-polyglot'
import ConnectorList from './ConnectorList'

const ConnectedList = ({ t, connectors, children }) => (
  <div className='content'>
    <h1>{t('connected title')}</h1>
    <ConnectorList connectors={connectors} />
    {children}
  </div>
)

export default translate()(ConnectedList)
