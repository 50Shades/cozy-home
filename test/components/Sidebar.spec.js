'use strict'

/* eslint-env jest */

import React from 'react'
import { shallow } from 'enzyme'

import { tMock } from '../jestLib/I18n'
import { Sidebar } from '../../src/components/Sidebar'

const routerMock = {
  location: {
    pathname: '/'
  }
}

const categoriesMock = ['isp', 'telecom']

describe('Sidebar component', () => {
  beforeEach(() => {
    jest.resetModules()
  })

  it('should be displayed correctly in root view', () => {
    const component = shallow(
      <Sidebar t={tMock} categories={categoriesMock} router={routerMock} />
    ).node
    expect(component).toMatchSnapshot()
  })

  it('should be displayed correctly in categories view', () => {
    const categoryRouter = Object.assign({}, routerMock, {
      location: {pathname: '/category/'}
    })
    const component = shallow(
      <Sidebar t={tMock} categories={categoriesMock} router={categoryRouter} />
    ).node
    expect(component).toMatchSnapshot()
  })
})
