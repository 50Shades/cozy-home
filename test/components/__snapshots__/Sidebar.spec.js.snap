exports[`Sidebar component should be displayed correctly in categories view 1`] = `
<aside
  className="coz-sidebar">
  <nav>
    <ul
      className="coz-nav">
      <li
        className="coz-nav-item">
        <Link
          activeClassName="active"
          className="coz-nav-link con-cat-discovery"
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/discovery">
          Collect by category
        </Link>
      </li>
      <li
        className="coz-nav-item">
        <Link
          className="coz-nav-link con-cat-categories active"
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/category/all">
          Providers
        </Link>
      </li>
      <ul
        className="con-nav-submenu">
        <li>
          <Link
            activeClassName="active"
            className="con-nav-submenu-link"
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/category/all">
            All
          </Link>
        </li>
        <li>
          <Link
            activeClassName="active"
            className="con-nav-submenu-link"
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/category/isp">
            ISP
          </Link>
        </li>
        <li>
          <Link
            activeClassName="active"
            className="con-nav-submenu-link"
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/category/telecom">
            Telecom
          </Link>
        </li>
        <li>
          <Link
            activeClassName="active"
            className="con-nav-submenu-link"
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/category/others">
            Others
          </Link>
        </li>
      </ul>
      <li
        className="coz-nav-item">
        <Link
          activeClassName="active"
          className="coz-nav-link con-cat-connected"
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/connected">
          Connected accounts
        </Link>
      </li>
    </ul>
  </nav>
</aside>
`;

exports[`Sidebar component should be displayed correctly in root view 1`] = `
<aside
  className="coz-sidebar">
  <nav>
    <ul
      className="coz-nav">
      <li
        className="coz-nav-item">
        <Link
          activeClassName="active"
          className="coz-nav-link con-cat-discovery"
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/discovery">
          Collect by category
        </Link>
      </li>
      <li
        className="coz-nav-item">
        <Link
          className="coz-nav-link con-cat-categories"
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/category/all">
          Providers
        </Link>
      </li>
      <li
        className="coz-nav-item">
        <Link
          activeClassName="active"
          className="coz-nav-link con-cat-connected"
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/connected">
          Connected accounts
        </Link>
      </li>
    </ul>
  </nav>
</aside>
`;
